src/Dependencies/Collections/SegmentedArray`1.cs

```diff
@@ -94,6 +94,39 @@ public SegmentedArray(int length)
             }
         }
 
+        public SegmentedArray<T> Resize(int newLength)
+        {
+            // For now, only allow growing resizes
+            if (newLength <= _length)
+                throw new ArgumentOutOfRangeException(nameof(newLength));
+
+            var newItems = new T[(newLength + SegmentSize - 1) >> SegmentShift][];
```

Comment:
>var newItems = new T[(newLength + SegmentSize - 1) >> SegmentShift][]; [](http://example.com/codeflow?start=12&length=70)

I considered using Array.Resize for this too, but in the case where the size doesn't change, it just returns the input array. Which we would then mutate. If we're fine modifying this SegmentedArray as part of this operation (which I think I might be) then this would simplify this a bit and improve the perf slightly.